### DISCUSSION TITLE
`writeContract` type inference

### DISCUSSION THREAD

**Discussion by dcposch** [wevm/viem#2386] [Question]
### Minimal example

Using latest viem: `2.13.8`
Typescript: `5.4.2`

Simplified for clarity:

```ts
const args: readonly [Hex, Hex, Hex] = [a, b, c];
const txHash = await vc.writeContract({ abi, address, functionName, args })
```

### Resulting type error

> **Type 'readonly [`0x${string}`, `0x${string}`, `0x${string}`]' is not assignable to type 'readonly [`0x${string}`, `0x${string}`, `0x${string}`]** & { blockExplorers?: { [x: string]: { name: string; url: string; apiUrl?: string | undefined; }; default: { name: string; url: string; apiUrl?: string | undefined; }; } | undefined; ... 10 more ...; serializers?: { ...; } | undefined; }'.

**Comment by jxom** [comment]
Can you please send a snippet of the ABI? or a minimal reproduction (more helpful)?

**Comment by dcposch** [comment]
all good, this was downstream of other type errors. solved, thanks
